dxr/plugins/js/tests/test_refs_and_needles/test_refs_and_needles.py

```diff
@@ -16,6 +16,10 @@ def test_needles(self):
                            "return <b>y</b>;", 9)
         self.found_line_eq("ref:foo",
                            "stuff.doStuff(<b>foo</b>);", 16)
+        self.found_line_eq("+id:Guy#fun",
+                           "this.<b>fun</b> = function thathasaname() {", 21)
```

Comment:
This is particularly impressive to me: that it realized `this` refers to the `new Guy`.